pkg/operation/botanist/secrets.go

```diff
@@ -338,16 +340,31 @@ func (b *Botanist) rotateSSHKeypairSecrets(ctx context.Context, gardenerResource
 	oldSecret.Name = v1beta1constants.SecretNameOldSSHKeyPair
 	gardenerResourceDataList.Upsert(oldSecret)
 
-	names := []string{
-		v1beta1constants.SecretNameSSHKeyPair,
-		v1beta1constants.SecretNameOldSSHKeyPair,
+	// current ssh keypair in shoot namespace in seed
+	secretSSHkeypair := &corev1.Secret{}
+	if err := b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, v1beta1constants.SecretNameSSHKeyPair), secretSSHkeypair); err != nil {
+		return err
 	}
 
-	for _, secretName := range names {
-		if err := b.K8sSeedClient.Client().Delete(ctx, &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: secretName, Namespace: b.Shoot.SeedNamespace}}); client.IgnoreNotFound(err) != nil {
-			return err
-		}
+	// old ssh keypair to be created in shoot namespace in seed
+	secretOldSSHkeypair := &corev1.Secret{
```

Comment:
Nit (symmetry):

```suggestion
	secretOldSSHKeyPair := &corev1.Secret{
```

Reply:
Addressed in [commit](https://github.com/gardener/gardener/pull/5388/commits/8e9970ad6576f23a3b8b79382aa9c8542ddd7ad4).